### DISCUSSION TITLE
Pass suffix to savers

### DISCUSSION THREAD

**Discussion by jcupitt** [libvips/libvips#2399] [Ideas]
See https://github.com/libvips/pyvips/issues/273

This code looks like it will write an AVIF file:

```python
image.write_to_target(buffer, '.avif')
```

But it will actually write HEIC, since `heifsave` defaults to HEIC, and without a filename to change that, `heifsave_target` will too. The suffix only selects the saver.

Proposal:

1. Add an optional `const char *suffix` arg to the save base class, perhaps `const char *options` too.
2. Set this from `vips_image_write_to_target()` / `vips_image_write_to_buffer()` /  `vips_image_write_to_file()`
3. Language bindings that reimplement these metafunctions would need the extra set too.
4. Operations like `heifsave_target` could check this to switch the default codec.

**Comment by jddcef** [comment]
Isn't this done already? https://github.com/libvips/libvips/discussions/3697#discussion-5697814
